events/include/events/EventQueue.h

```diff
@@ -89,30 +89,28 @@ class EventQueue : private mbed::NonCopyable<EventQueue> {
     /** Dispatch events
      *
      *  Executes events until the specified milliseconds have passed.
-     *  If ms is negative, the dispatch function will dispatch events
-     *  indefinitely or until break_dispatch is called on this queue.
      *
-     *  When called with a finite timeout, the dispatch function is guaranteed
-     *  to terminate. When called with a timeout of 0, the dispatch function
-     *  does not wait and is IRQ safe.
+     *  The dispatch function is guaranteed to terminate after the elapsed wait
      *
-     *  @param ms       Time to wait for events in milliseconds, a negative
-     *                  value will dispatch events indefinitely
-     *                  (default to -1)
+     *  @param wait     Time to wait for events in milliseconds, expressed as a 
+     *                  Chrono duration.
      */
-    void dispatch(int ms = -1);
+    void dispatch(duration wait);
 
     /** Dispatch events without a timeout
-     *
-     *  This is equivalent to EventQueue::dispatch with no arguments, but
-     *  avoids overload ambiguities when passed as a callback.
-     *
-     *  @see EventQueue::dispatch
+     * 
+     * Executes events indefinitely unless the dispatch loop is forcibly broken.
+     * See break_dispatch()
+     * 
      */
-    void dispatch_forever()
-    {
-        dispatch();
-    }
+    void dispatch_forever();
+
+    /** Dispatch currently queued events only and then terminate
+     * 
+     * In this case, the dispatch function does not wait and is IRQ safe.
+     * 
+     */
+    void dispatch_once();
```

Comment:
I lost my comment for that item. 

@kjbracey-arm The implementation doesn't do exactly this. It may not break immediately out of the loop if the next tick hasn't been reached. Do you think we should patch the implementation ? 
The change is trivial, we just have to return if `ms` is equal to 0.  

https://github.com/ARMmbed/mbed-os/blob/68612cb1e7e964b8704f020212204d168f73f09d/events/source/equeue.c#L525-L530

Reply:
I'm not seeing what you're seeing here. If `ms == 0`, then it computes `tickdiff(timeout, tick)`, ie `entry_tick + 0 - tick`, and that's got to be `<= 0`, so it exits immediately. Right?

BTW the name here feels perilously close to `dispatch_one()`, but I guess it's okay...